packages/ignite/src/commands/plugin.js

```diff
@@ -0,0 +1,145 @@
+ * 
+ * @param {Object} context The gluegun context.
+ * @returns {Object} The answers.
+ */
+const walkthrough = async (context) => {
+  const minOptions = { template: 'No', command: 'No' }
+  const maxOptions = { template: 'Yes', command: 'Yes' }
+  if (context.parameters.options.min) { return minOptions }
+  if (context.parameters.options.max) { return maxOptions }
+
+  // Okay, we'll ask one by one, fine
+  return await context.prompt.ask([
+    {
+      name: 'template',
+      message: 'Would you like to generate an example component?',
+      type: 'list',
+      choices: ['No', 'Yes']
+    },
+    {
+      name: 'command',
+      message: 'Would you like to generate an example command/generator?',
+      type: 'list',
+      choices: ['No', 'Yes']
+    }
+  ])
+}
+
+/**
+ * Checks whether a plugin name was given and errors if not.
+ *
+ * @param {Object} context The gluegun context.
+ * @returns {Boolean}
+ */
+const checkName = (pluginName, context) => {
+  const { strings, print, runtime, parameters } = context
+  
+  if (strings.isBlank(pluginName)) {
+    print.info(`${runtime.brand} plugin new <${runtime.brand}-plugin-name>\n`)
+    print.error('Plugin name is required')
+    process.exit(exitCodes.PLUGIN_NAME)
+    return false
+  }
+
+  if (pluginName.indexOf('ignite-') !== 0) {
+    // As passive-aggressively as possible
+    print.info(`Ignite plugins typically start with ignite-* -- but you do you.`)
+  }
+
+  // TODO: Make this better at detecting invalid plugin names
+  if (!/^[a-z0-9].*/i.test(pluginName)) {
+    print.error('Plugin name should be a valid folder name')
+    process.exit(exitCodes.PLUGIN_NAME)
+    return false
+  }
+
+  return true
+}
+
+/**
+ * Creates a new folder with the plugin files.
+ *
+ * @param {Object} context The gluegun context.
+ */
+const createNewPlugin = async (context) => {
+  const { parameters, print, ignite, strings, filesystem } = context
+  const pluginName = parameters.third
+  const name = strings.pascalCase(pluginName.replace(/^ignite-/, ""))
+
+  // Validate plugin name
+  if (!checkName(pluginName, context)) { return }
+
+  // Plugin generation walkthrough
+  const answers = await walkthrough(context)
+
+  // Here we go!
+  print.info(`Creating new plugin: ${pluginName}`)
+
+  const copyJobs = [
+    { template: 'plugin/.gitignore', target: `${pluginName}/.gitignore` },
+    { template: 'plugin/index.js.ejs', target: `${pluginName}/index.js` },
+    { template: 'plugin/ignite.toml.ejs', target: `${pluginName}/ignite.toml` },
+    { template: 'plugin/package.json.ejs', target: `${pluginName}/package.json` },
+    (answers.template === 'Yes') &&
+      { template: 'plugin/templates/Example.js.ejs', target: `${pluginName}/templates/${name}Example.js` },
+    (answers.command === 'Yes') &&
+      { template: 'plugin/commands/example.js.ejs', target: `${pluginName}/commands/example.js` },
+  ]
+
+  // copy over the files
+  await ignite.copyBatch(context, copyJobs, {name, pluginName})
```

Comment:
I think maybe gluegun's `context.template.generate` (and `context.ignite.generate`) should support this variant too. Instead of `copyBatch`, we could call `ignite.generate()` with the same arguments.